src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -65,9 +65,6 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
                 throw new ArgumentNullException(nameof(bindingContext));
             }
 
-            // For compatibility with MVC 5.0 for top level object we want to consider an empty key instead of
-            // the parameter name/a custom name. In all other cases (like when binding body to a property) we
-            // consider the entire ModelName as a prefix.
```

Comment:
This comment still applies because errors the input formatter adds still use this prefix.

Further, conversion errors e.g. during XML input formatting can still lead to a `ModelState` entry w/ `Key == string.Empty`. That probably isn't what we want.

How about removing the `string.Empty` special case entirely? That is, removing the value-only `ModelState` entry doesn't clean up all `ModelState` entries this binder may add.